lightkurve/seismology/numax_estimators.py

```diff
@@ -0,0 +1,155 @@
+
+    #C: What is this doing? Why have these values been picked? This function is slow.
+    if window is None:
+        if u.Quantity(periodogram.frequency[-1], u.microhertz) > u.Quantity(500., u.microhertz):
+            window = u.Quantity(250., u.microhertz).to(periodogram.frequency.unit).value
+        else:
+            window = u.Quantity(25., u.microhertz).to(periodogram.frequency.unit).value
+
+    # Calculate the spacing size
+    if spacing is None:
+        if u.Quantity(periodogram.frequency[-1], u.microhertz) > u.Quantity(500., u.microhertz):
+            spacing = u.Quantity(10., u.microhertz).to(periodogram.frequency.unit).value
+        else:
+            spacing = u.Quantity(1., u.microhertz).to(periodogram.frequency.unit).value
+
+    # Run some checks on the inputs
+    window = u.Quantity(window, periodogram.frequency.unit).value
+    spacing = u.Quantity(spacing, periodogram.frequency.unit).value
+    if numaxs is None:
+        numaxs = np.arange(np.ceil(np.nanmin(periodogram.frequency.value)) + window/2,
+                    np.floor(np.nanmax(periodogram.frequency.value)) - window/2,
+                    spacing)
+    numaxs = u.Quantity(numaxs, periodogram.frequency.unit).value
+    if not hasattr(numaxs, '__iter__'):
+        numaxs = np.asarray([numaxs])
+
+    fs = np.median(np.diff(periodogram.frequency.value))
+    for var, label in zip([np.asarray(window), np.asarray(spacing), numaxs],
+                          ['window', 'spacing', 'numaxs']):
+        if (var < fs).any():
+            raise ValueError("You can't have {} smaller than the "
+                            "frequency separation!".format(label))
+        if (var > (periodogram.frequency[-1].value - periodogram.frequency[0].value)).any():
+            raise ValueError("You can't have {} wider than the entire "
+                            "power spectrum!".format(label))
+        if (var < 0).any():
+            raise ValueError("Please pass an entirely positive {}.".format(label))
+
+    # We want to find the numax which returns in the highest autocorrelation
+    # power, rescaled based on filter width
+    fs = np.median(np.diff(periodogram.frequency.value))
+
+    metric = np.zeros(len(numaxs))
+    acf2d = np.zeros([int(window/2/fs)*2,len(numaxs)])
+    for idx, numax in enumerate(numaxs):
+        acf = utils.autocorrelate(periodogram, numax, window=window, frequency_spacing=fs)      #Return the acf at this numax
+        acf2d[:, idx] = acf                                     #Store the 2D acf
+        metric[idx] = (np.sum(np.abs(acf)) - 1 ) / len(acf)  #Store the max acf power normalised by the length
+
+    # Smooth the data to find the peak
+    # Previous smoothing could be completely wrong, it's based on the length of the array, not the frequency!!!
+    # It needs to be based on the frequency differences in `numaxs`
+    if len(numaxs) > 10:
+        g = Gaussian1DKernel(stddev=100 * np.nanmedian(np.diff(numaxs)))
```

Comment:
This results in the method failing to recover numax for MS stars. Since the Gaussian1DKernel just takes a stddev in indices and doesn't care about the x-axis units, shouldn't the previous method still work?